aiida/restapi/common/utils.py

```diff
@@ -146,63 +146,79 @@ def parse_path(self, path_string):
         :param path_string: the path string
         :return:resource_type (string)
                 page (integer)
-                pk (integer)
+                id (integer or uuid)
                 result_type (string))
         """
+        import uuid
 
         ## Initialization
         page = None
-        pk = None
+        id = None
         query_type = "default"
         path = self.split_path(self.strip_prefix(path_string))
 
         ## Pop out iteratively the "words" of the path until it is an empty
         # list.
         ##  This way it should be easier to plug in more endpoint logic
+
         # Resource type
         resource_type = path.pop(0)
         if not path:
-            return (resource_type, page, pk, query_type)
```

Comment:
The new logic of `load_node` is that if the parameter is a string, it is considered the start of a UUID; if integer, a PK.
When coming from the REST request, it's always a string. So this rises an ambiguity: if I want to pass just the beginning of the following uuid:
```
183672693a7ab39abc6adfe03...
```
I could pass ``1836`` for instance, and it is not clear if it's a pk or the start of the uuid.

Is it ok to implement only uuid or its start? Or the PK is still needed? in this case, by the way, in line 175 I wouldn't use the uuid.UUID function, but just pass the string?